### PR TITLE
:sparkles: File Download button for multi fence aware file repo

### PR DESCRIPTION
**Pull request by joneubank**

### CODE REVIEW

File: src/components/FileRepo/FileRepo.js [hunk 1]
```diff
@@ -92,190 +90,186 @@ const TableHeaderContent = ({ sqon, disabled, ...props }) => {
...
+        effects,
+        api,
+        state: { fenceConnectionsInitialized },
+      } = this.props;
+      // Only fetch connections once - don't fetch if we've done it previously
+      if (!fenceConnectionsInitialized) {
+        effects.fetchFenceConnections({ api });
+      }
+    },
+  }),
```
Comment (hlminh2000) [inline src/components/FileRepo/FileRepo.js]: can use our new hoc now :)

Comment (joneubank) [reply src/components/FileRepo/FileRepo.js]: si

File: src/stateProviders/provideFenceConnections.js [hunk 2]
```diff
@@ -46,7 +49,21 @@ export default provideState({
...
+            return getUserStudyPermission(api, {
+              [fence]: details,
+            })({})
+              .then(({ acceptedStudiesAggs, unacceptedStudiesAggs }) =>
+                effects.addFenceStudies(fence, {
+                  authorizedStudies: acceptedStudiesAggs,
+                  unauthorizedStudies: unacceptedStudiesAggs,
+                }),
+              )
+              .catch(err => console.log(`Error fetching fence studies for '${fence}': ${err}`));
```
Comment (hlminh2000) [inline src/stateProviders/provideFenceConnections.js]: if one fails, the `Promise.all` will reject. I the dashboard card isn't able to handle that eh

Comment (joneubank) [reply src/stateProviders/provideFenceConnections.js]: 👍

**Pull request merged**